libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_ota.h

```diff
@@ -0,0 +1,120 @@
+#include <azure/core/az_json.h>
+#include <azure/iot/az_iot_hub_client.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/* ADU PnP Component Name */
+#define AZ_IOT_ADU_PROPERTIES_COMPONENT_NAME                 "deviceUpdate"
+
+/* ADU Service Actions */
+#define AZ_IOT_ADU_OTA_SERVICE_ACTION_APPLY_DEPLOYMENT       3
+#define AZ_IOT_ADU_OTA_SERVICE_ACTION_CANCEL                 255
+
+/* ADU Agent States */
+#define AZ_IOT_ADU_OTA_AGENT_STATE_IDLE                      0
+#define AZ_IOT_ADU_OTA_AGENT_STATE_DEPLOYMENT_IN_PROGRESS    6
+#define AZ_IOT_ADU_OTA_AGENT_STATE_FAILED                    255
+
+
+/* Maximum Number of Files Handled by this ADU Agent */
+#define AZ_IOT_ADU_OTA_FILE_URL_MAX_COUNT 10
+
+typedef struct
+{
+    az_span manufacturer;
+    az_span model;
+    az_span last_installed_update_id;
+    az_span adu_version;
+    az_span do_version;
+} az_iot_adu_ota_device_information;
+
+typedef struct
+{
+    struct
+    {
+        int32_t action;
+        az_span id;
+        az_span retry_timestamp;
+    } _internal;
+} az_iot_adu_ota_workflow;
+
+typedef struct
+{
+    struct
+    {
+        az_span id;
+        az_span url;
+    } _internal;
+} az_iot_adu_ota_file_url;
+
+typedef struct
+{
+  struct
+  {
+    /* Device Qualifications */
+    az_iot_hub_client const* client;
+    az_iot_adu_ota_device_information device_information;
+
+    /* ADU Service Properties */
+    az_iot_adu_ota_workflow workflow;
+    az_span update_manifest;
+    az_span update_manifest_signature;
+    az_iot_adu_ota_file_url file_urls[AZ_IOT_ADU_OTA_FILE_URL_MAX_COUNT];
+    uint32_t file_urls_count;
+  } _internal;
+} az_iot_adu_ota_state; 
+
+AZ_NODISCARD bool az_iot_adu_ota_is_component_device_update(
+    az_span component_name);
+
+AZ_NODISCARD az_result az_iot_adu_ota_init(az_iot_adu_ota_state* ota_state,
+    az_iot_hub_client const* client,
+    az_iot_adu_ota_device_information device_information
+    );
+
+AZ_NODISCARD az_result az_iot_adu_ota_get_properties_payload(
+    az_iot_adu_ota_state* ota_state, int32_t agent_state, az_span payload, az_span* out_payload);
+
+AZ_NODISCARD az_result az_iot_adu_ota_parse_service_properties(
+    az_iot_adu_ota_state* ota_state, az_json_reader* jr, az_span buffer, az_span* buffer_remainder);
+
+AZ_NODISCARD az_result az_iot_adu_ota_get_workflow(az_iot_adu_ota_state* ota_state, az_span* id, int32_t* action, az_span* retry_timestamp);
+
+AZ_NODISCARD az_result az_iot_adu_ota_get_update_manifest(az_iot_adu_ota_state* ota_state, az_span* update_manifest);
+
+AZ_NODISCARD az_result az_iot_adu_ota_get_update_manifest_signature(az_iot_adu_ota_state* ota_state, az_span* update_manifest_signature);
+
+AZ_NODISCARD az_result az_iot_adu_ota_get_file_url_count(az_iot_adu_ota_state* ota_state, uint32_t* file_url_count);
+
+AZ_NODISCARD az_result az_iot_adu_ota_get_file_url(az_iot_adu_ota_state* ota_state, uint32_t index, az_span* file_id, az_span* file_url);
```

Comment:
nit of nits: I would flip these (order).

Reply:
These "get" functions are now gone.